### PR TITLE
Create Avatar Image and Fix Emoji Picker

### PR DESCRIPTION
**Pull request by wojtus7**
https://linear.app/rainbow/issue/RAI-628/get-avatar-builder-working-again-photo-selection

Zrzut ekranu 2020-07-24 o 18 29 28

To test it out please change flag `@objc var isAvatarPickerAvailable: Bool = false` in `TransactionListView.swift`

### CODE REVIEW

File: ios/TransactionListView.swift [hunk 1]
```diff
@@ -57,7 +57,23 @@ class TransactionListView: UIView, UITableViewDelegate, UITableViewDataSource {
       header.accountNameViewWidthConstraint.constant = accountAddressWidth
     }
   }
-  
+  @objc var accountImage: NSString? = nil {
+   didSet {
+    if (accountImage != nil) {
+      let url = URL.init(fileURLWithPath: accountImage?.expandingTildeInPath ?? "")
```
Comment (osdnk) [inline ios/TransactionListView.swift:63]: Assuming the account image is not `nil`
```suggestion
      let url = URL.init(fileURLWithPath: accountImage!.expandingTildeInPath)
```

File: ios/TransactionListView.swift [hunk 2]
```diff
@@ -192,7 +206,7 @@ class TransactionListView: UIView, UITableViewDelegate, UITableViewDataSource {
     header.accountView.isHidden = false
     
     header.addSubview(headerSeparator)
-    if(isAvatarPickerAvailable){
+    if(true){
```
Comment (osdnk) [inline ios/TransactionListView.swift:209]:
```suggestion
    if(isAvatarPickerAvailable){
```

:)

**Comment by christianbaroni** [comment]
hey @wojtus7 !

just tested this, few things I noticed:

- selecting red avatar color doesn't work
- when you add an emoji, then go to edit the name of the wallet, the emoji shows as part of the string. it shouldn't
- isAvatarPickerAvailable doesn't work (I rebased over develop locally, maybe that's why). only way I could enable was by manually setting to true in TransactionListView.swift

can we also split this into two flags, isAvatarEmojiPickerEnabled and isAvatarImagePickerEnabled. if only avatar picker is enabled, it should skip the action sheet and go straight to the emoji picker

otherwise this looks great!

**Review by christianbaroni — DISMISSED** [review]
hey @wojtus7 !

just tested this, few things I noticed:

- [x] selecting red avatar color doesn't work
- [x] when you add an emoji, then go to edit the name of the wallet, the emoji shows as part of the string. it shouldn't
- [x] isAvatarPickerAvailable doesn't work (I rebased over develop locally, maybe that's why). only way I could enable was by manually setting to true in TransactionListView.swift
- [x] no worries if not, but is there an easy way to change the crop square to a circle when choosing an image?
- [x] can we also split this into two flags, isAvatarEmojiPickerEnabled and isAvatarImagePickerEnabled? if only avatar picker is enabled, it should skip the action sheet and go straight to the emoji picker

otherwise this looks great!

**Comment by wojtus7** [comment]
@christianbaroni can you take a look? 

I think all things mentioned above are fixed now 🎉

**Comment by christianbaroni** [comment]
this is good to code review @mikedemarais

confirmed working as expected

File: ios/TransactionListView.swift [hunk 3]
```diff
@@ -109,6 +129,12 @@ class TransactionListView: UIView, UITableViewDelegate, UITableViewDataSource {
     }
   }
   
+  @objc func onPressInAvatar(_ sender: UIButton) {
```
Comment (osdnk) [inline ios/TransactionListView.swift:132]: Can't you reuse the logic from UIView animations?

File: src/navigation/effects.js [hunk 4]
```diff
@@ -267,7 +267,7 @@ export const emojiPreset = {
         pointerEvents="none"
         style={{
           backgroundColor: 'rgb(51, 54, 59)',
-          height: deviceUtils.dimensions.height,
+          height: deviceUtils.dimensions.height + 50,
```
Comment (osdnk) [inline src/navigation/effects.js:270]: the animation of the modal can be faster I think. But it's up to @christianbaroni

**Review by osdnk — APPROVED** [review]
Two v v v v v v minors https://share.icloud.com/photos/0Fvkt6yuPOLWvzBrtciABGuZw

1. headers are switching and jumping up and down a bit
2. While changing the color selector maybe we can animate color's transition?

Probably it's not worth blocking

**Comment by christianbaroni** [comment]
hey @wojtus7 one more small thing, can you add a 'Remove Photo' button to the action sheet?

**Comment by mikedemarais** [comment]
> hey @wojtus7 one more small thing, can you add a 'Remove Photo' button to the action sheet?

yo @wojtus7 ill handle this!

**Review by mikedemarais — APPROVED** [review]

**Pull request merged**